### PR TITLE
[SECTY-563] feat(st-two-list): not show check all if list is empty. Apply new styles.

### PR DESCRIPTION
**Pull request by jdiaz-stratio**
### Documentation
- [x] Add the issue in PR description
- [x] Have changelog updated
- [x] You fill the milestone value
- [x] You add some label
- [x] Have example running in demo app
- [x] Review id on demo is the same of egeo folder

### Code
- [x] Pass Sass lint task
- [ ] Have qaTags

### Tests
- [x] Have at least 70% tests coverage

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## 13.0.0 (upcoming)
 
+**New features:**
+
+* st-two-list: Show check all button only if it has elements and applies onClick only in st-checkbox.
```
Comment (emgaitan-stratio) [inline CHANGELOG.md]: Added click listener only to checkbox and 'check all' button is only  displayed when it has elements.

Comment (jdiaz-stratio) [reply CHANGELOG.md]: Done

File: src/lib/st-two-list-selection/list-selection/list-selection.component.html [hunk 2]
```diff
@@ -11,9 +11,14 @@
 
 -->
 <div class="list-selection">
-   <p class="title sth-two-list__list-title" [ngClass]="{'sth-two-list__important-title': important}">{{title}}</p>
-   <st-search class="search" *ngIf="hasSearch && ( list && list.length >= showSearchNumber) || (hasSearch && !showSearchNumber)" [placeholder]="searchPlaceholder" [qaTag]="searchQaTag" (search)="search.emit($event)"></st-search>
+   <div class="list-selection__rectangle">
+      <p class="title sth-two-list__list-title" [ngClass]="{'sth-two-list__important-title': important}">{{title}}</p>
+      <p class="subtitle sth-two-list__list-subtitle">{{subtitle}}</p>
+      <st-search [disabled]="list && list.length === 0" class="search" *ngIf="hasSearch && ( list && list.length >= showSearchNumber) || (hasSearch && !showSearchNumber)" [placeholder]="searchPlaceholder" [qaTag]="searchQaTag" (search)="search.emit($event)"></st-search>
```
Comment (emgaitan-stratio) [inline src/lib/st-two-list-selection/list-selection/list-selection.component.html]: When list is undefined, search will not be disabled

File: src/lib/st-two-list-selection/list-selection/list-selection.component.html [hunk 3]
```diff
@@ -11,9 +11,14 @@
...
+   </div>
    <list-scroll [list]="list" [itemAll]="itemAll" [mode]="mode" (selectItem)="selectItem.emit($event)" [hasAllList]="hasAllList" [qaTag]="listQaTag" [editable]="editable" [qaTag]="listQaTag" [editable]="editable"
              (selectExtraLabel)="selectExtraLabel.emit($event)"></list-scroll>
-   <st-select *ngIf="hasOrder" class="orderCombo" button="Order" [options]="orderOptions" (select)="changeOrder.emit($event)"></st-select>
+   <div class="list-selection__rectangle bottom" *ngIf="hasOrder" >
+         <st-select [disabled]="list && list.length === 0" class="orderCombo" button="Order" [options]="orderOptions" (select)="changeOrder.emit($event)" [placeholder]="orderPlaceholder"></st-select>
```
Comment (emgaitan-stratio) [inline src/lib/st-two-list-selection/list-selection/list-selection.component.html]: When list is undefined, select will not be disabled

Change "orderCombo" class to "order-select"

File: src/lib/st-two-list-selection/list-selection/list-selection.component.scss [hunk 4]
```diff
@@ -35,12 +59,11 @@
 }
 
 .search {
-   margin: 10px 0;
+   margin-top: 18px;
    display: block;
 }
 
 .orderCombo {
```
Comment (emgaitan-stratio) [inline src/lib/st-two-list-selection/list-selection/list-selection.component.scss]: Better order-select

File: src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts [hunk 5]
```diff
@@ -52,8 +62,28 @@ export class StTwoListSelectionDemoComponent {
...
+
+      if (this.selectedUserList.length === 0) {
+         this.configWithCheckAll.selectedElementsListSubtitle = `0 users`;
+      } else {
+         this.configWithCheckAll.selectedElementsListSubtitle = `0/${this.selectedUserList.length} users selected`;
+      }
+   }
+
+   onNumItemsAll(event: Event): void {
+      console.log('Number of elements selected', event);
```
Comment (emgaitan-stratio) [inline src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts]: Remove console log

File: src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts [hunk 6]
```diff
@@ -52,8 +62,28 @@ export class StTwoListSelectionDemoComponent {
...
+      this.configWithCheckAll.allElementsListSubtitle = `${event}/${this.completeUserList.length} users selected`;
+   }
+
    onNumItemsSelected(event: Event): void {
       console.log('Number of elements selected', event);
```
Comment (emgaitan-stratio) [inline src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts]: Remove console log

File: src/lib/st-two-list-selection/list-selection/list-selection.component.html (hunk 2 above)
Comment (jdiaz-stratio) [reply src/lib/st-two-list-selection/list-selection/list-selection.component.html]: Done

File: src/lib/st-two-list-selection/list-selection/list-selection.component.html (hunk 3 above)
Comment (jdiaz-stratio) [reply src/lib/st-two-list-selection/list-selection/list-selection.component.html]: Done

File: src/lib/st-two-list-selection/list-selection/list-selection.component.scss (hunk 4 above)
Comment (jdiaz-stratio) [reply src/lib/st-two-list-selection/list-selection/list-selection.component.scss]: Done

File: src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts (hunk 5 above)
Comment (jdiaz-stratio) [reply src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts]: Done

File: src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts (hunk 6 above)
Comment (jdiaz-stratio) [reply src/egeo-demo/st-two-list-selection-demo/st-two-list-selection-demo.ts]: Done

**Pull request merged**